### PR TITLE
Use Kontena::Command hooks and new authentication mechanism.

### PR DESCRIPTION
**Pull request by kke**

### CODE REVIEW

File: lib/kontena/machine/upcloud/cloudinit_master.yml [hunk 1]
```diff
@@ -89,8 +89,11 @@ coreos:
             --link kontena-server-mongo:mongodb \
             -e MONGODB_URI=mongodb://mongodb:27017/kontena_server \
 <% end -%>
-<% if auth_server %>
-            -e AUTH_API_URL=<%= auth_server %> \
+<% if server_name %>
```
Comment (jakolehm) [inline lib/kontena/machine/upcloud/cloudinit_master.yml]: No newlines allowed

File: lib/kontena/machine/upcloud/cloudinit_master.yml [hunk 2]
```diff
@@ -89,8 +89,11 @@ coreos:
...
+            -e KONTENA_SERVER_NAME=<%= server_name %> \
```
Comment (jakolehm) [inline lib/kontena/machine/upcloud/cloudinit_master.yml]: `KONTENA_SERVER_NAME` => `SERVER_NAME`

File: lib/kontena/machine/upcloud/cloudinit_master.yml [hunk 3]
```diff
@@ -89,8 +89,11 @@ coreos:
...
+<% end -%>
+<% if initial_admin_code %>
+            -e KONTENA_INITIAL_ADMIN_CODE=<%= initial_admin_code %> \
```
Comment (jakolehm) [inline lib/kontena/machine/upcloud/cloudinit_master.yml]: `KONTENA_INITIAL_ADMIN_CODE` => `INITIAL_ADMIN_CODE`

File: lib/kontena/machine/upcloud/master_provisioner.rb [hunk 4]
```diff
@@ -83,7 +80,7 @@ def run!(opts)
             }
           }.to_json
 
-          ShellSpinner "Creating Upcloud master #{hostname.colorize(:cyan)} " do
+          spinner "* Creating Upcloud master #{hostname.colorize(:cyan)} " do
```
Comment (jakolehm) [inline lib/kontena/machine/upcloud/master_provisioner.rb]: why `*` ?

**Review by jakolehm — CHANGES_REQUESTED** [review]

**Comment by kke** [comment]
Fixes applied

**Review by jakolehm — APPROVED** [review]

**Pull request merged**